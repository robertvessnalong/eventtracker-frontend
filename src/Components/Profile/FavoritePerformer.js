import { useContext, useState, useEffect } from 'react';
import UserContext from '../../Context/UserContext';
import { Row, Col, Spinner, Card } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import uuid from 'react-uuid';

const FavoritePerformer = () => {
  const { user } = useContext(UserContext);
  const [favorites, setFavorites] = useState([]);
  const [loading, isLoading] = useState(true);

  useEffect(() => {
    const setUserFavorites = () => {
      isLoading(false);
      const performers = user.favorites.filter(
        ({ type }) => type === 'performer'
      );
      setFavorites(performers);
    };
    if (user.favorites) {
      setUserFavorites();
    }
  }, [user.favorites]);

  if (loading) {
    return (
      <Row>
        <Col>
          <Spinner animation='border' role='status'>
            <span className='visually-hidden'>Loading...</span>
          </Spinner>
        </Col>
      </Row>
    );
  }

  if (favorites.length < 1) {
    return (
      <Row>
        <Col>
          <span>You do not have any favorite performers yet!</span>
        </Col>
      </Row>
    );
  }

  return (
    <Row>
      <Col>
        {favorites.map((favorite) => (
          <Card key={uuid()} className='Favorite mb-4'>
            <Card.Body className='d-flex flex-row justify-content-between align-items-center'>
              <Card.Text as='div'>{favorite.performerName}</Card.Text>
              <Link
                className='blue-btn d-inline'
                to={`/performers/${favorite.performerId}`}
              >
                See Events
              </Link>
            </Card.Body>
          </Card>
        ))}
      </Col>
    </Row>
  );
};

export default FavoritePerformer;
