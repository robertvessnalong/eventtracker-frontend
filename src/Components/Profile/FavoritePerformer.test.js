import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import FavoritePerformer from './FavoritePerformer';
import TestUser from '../../Testing/TestUser';
import UserContext from '../../Context/UserContext';
import { BrowserRouter } from 'react-router-dom';

describe('Favorite Performer Component', () => {
  // Smoke Test
  test('renders without crashing', async () => {
    render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <FavoritePerformer />
        </UserContext.Provider>
      </BrowserRouter>
    );

    const label = screen.getByText(
      /College Football Playoff National Championship Game/i
    );
    expect(label).toBeInTheDocument();
  });

  // Snapshot
  test('matches snapshot', async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <FavoritePerformer />
        </UserContext.Provider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
