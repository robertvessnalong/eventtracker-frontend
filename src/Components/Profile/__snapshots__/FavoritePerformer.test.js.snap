// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Performer Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="Favorite mb-4 card"
      >
        <div
          class="d-flex flex-row justify-content-between align-items-center card-body"
        >
          <div
            class="card-text"
          >
            College Football Playoff National Championship Game
          </div>
          <a
            class="blue-btn d-inline"
            href="/performers/12007"
          >
            See Events
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
